src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -141,6 +143,29 @@ internal WebApplicationBuilder(WebApplicationOptions options, Action<IHostBuilde
         /// <returns>A configured <see cref="WebApplication"/>.</returns>
         public WebApplication Build()
         {
+            // The `WebRoot` is the only host configuration that can be modified after
+            // the `WebApplicationBuilder` is instantiated. When the `WebRoot` is set
+            // we want to create a matching file provider that is used in DI to resolve
+            // the full paths for static assets in the application. We do this in the
+            // `Build` invocation because we want to respect only the last of the
+            // WebRootPaths configured on the application. So, when the following code
+            // is called, a directory and provider will only be created for C.
+            // builder.WebHost.UseSettings("WEBROOT", "A");
+            // builder.WebHost.UseSettings("WEBROOT", "B");
+            // builder.WebHost.UseSettings("WEBROOT", "C");
+            if (!string.IsNullOrEmpty(Environment.WebRootPath))
+            {
+                Directory.CreateDirectory(Environment.WebRootPath);
+                Environment.WebRootFileProvider = new PhysicalFileProvider(Environment.WebRootPath);
+                // The StaticWebAssets loader wraps the WebRootFileProvider in order to support loading
+                // static assets from a manifest file and from a local directory. Since we modified
+                // the WebRootFileProvider, we call the StaticWebAssetsLoader to update.
+                if (Environment.IsDevelopment())
+                {
+                    StaticWebAssetsLoader.UseStaticWebAssets(Environment, Configuration);
```

Comment:
Good catch! Is there a test that fails if we forget to do this? If not, it'd be nice to add.

Reply:
Yep -- the template tests failing to resolve scoped CSS was what helped catch this.